libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomSummaryDetailsFactory.kt

```diff
@@ -28,10 +29,19 @@ import org.matrix.rustcomponents.sdk.RoomListItem
 import org.matrix.rustcomponents.sdk.use
 
 class RoomSummaryDetailsFactory(private val roomMessageFactory: RoomMessageFactory = RoomMessageFactory()) {
-    suspend fun create(roomListItem: RoomListItem): RoomSummaryDetails {
+    suspend fun create(roomListItem: RoomListItem, roomSyncSubscriber: RoomSyncSubscriber): RoomSummaryDetails {
         val roomInfo = roomListItem.roomInfo()
-        val latestRoomMessage = roomListItem.latestEvent()?.use {
-            roomMessageFactory.create(it)
+        val latestRoomMessage = roomListItem.latestEvent().use { event ->
+            // If there is no latest event, subscribe to the room to get the latest 20 events, then retry
+            val latestEvent = event ?: run {
```

Comment:
Do we really want do to that? Are we not waiting for the sdk to do his own auto pagination?

Reply:
This should be a temporary workaround until the SDK does this automatically.

Reply:
Could we add this in a separate PR then?

Reply:
Sure 👍

Reply:
Removed https://github.com/element-hq/element-x-android/pull/3145/commits/342c1cf4a630d37be25b7b1c75ff54ac6bd48564 from this branch.